src/test/java/com/conveyal/gtfs/loader/JDBCTableWriterTest.java

```diff
@@ -140,6 +144,171 @@ public void canPreventSQLInjection() throws IOException, SQLException, InvalidNa
         assertThat(createdFeedInfo.feed_publisher_name, equalTo(publisherName));
     }
 
+    @Test
+    public void canCreateUpdateAndDeleteFares() throws IOException, SQLException, InvalidNamespaceException {
+        // Store Table and Class values for use in test.
+        final Table fareTable = Table.FARE_ATTRIBUTES;
+        final Class<FareDTO> fareDTOClass = FareDTO.class;
+
+        // create new object to be saved
+        FareDTO fareInput = new FareDTO();
+        String fareId = "2A";
+        fareInput.fare_id = fareId;
+        fareInput.currency_type = "USD";
+        fareInput.price = 2.50;
+        fareInput.agency_id = "RTA";
+        fareInput.payment_method = 0;
+        // Empty value should be permitted for transfers and transfer_duration
+        fareInput.transfers = null;
+        fareInput.transfer_duration = null;
+        FareRuleDTO fareRuleInput = new FareRuleDTO();
+        // Fare ID should be assigned to "child entity" by editor automatically.
+        fareRuleInput.fare_id = null;
+        fareRuleInput.route_id = null;
+        // FIXME There is currently no check for valid zone_id values in contains_id, origin_id, and destination_id.
+        fareRuleInput.contains_id = "any";
+        fareRuleInput.origin_id = "value";
+        fareRuleInput.destination_id = "permitted";
+        fareInput.fare_rules = new FareRuleDTO[]{fareRuleInput};
+
+        // convert object to json and save it
+        JdbcTableWriter createTableWriter = createTestTableWriter(fareTable);
+        String createOutput = createTableWriter.create(mapper.writeValueAsString(fareInput), true);
+        LOG.info("create {} output:", fareTable.name);
+        LOG.info(createOutput);
+
+        // parse output
+        FareDTO createdFare = mapper.readValue(createOutput, fareDTOClass);
+
+        // make sure saved data matches expected data
+        assertThat(createdFare.fare_id, equalTo(fareId));
+        assertThat(createdFare.fare_rules[0].fare_id, equalTo(fareId));
+
+        // try to update record
+        String updatedFareId = "3B";
+        createdFare.fare_id = updatedFareId;
+
+        // covert object to json and save it
+        JdbcTableWriter updateTableWriter = createTestTableWriter(fareTable);
+        String updateOutput = updateTableWriter.update(
+                createdFare.id,
+                mapper.writeValueAsString(createdFare),
+                true
+        );
+        LOG.info("update {} output:", fareTable.name);
+        LOG.info(updateOutput);
+
+        FareDTO updatedFareDTO = mapper.readValue(updateOutput, fareDTOClass);
+
+        // make sure saved data matches expected data
+        assertThat(updatedFareDTO.fare_id, equalTo(updatedFareId));
+        assertThat(updatedFareDTO.fare_rules[0].fare_id, equalTo(updatedFareId));
```

Comment:
We should also be testing that the value was updated in the database in these tests.

Reply:
Referencing https://github.com/conveyal/gtfs-lib/issues/147

Reply:
Resolving this now that the ticket has been created.